datajoint/table.py

```diff
@@ -344,23 +348,43 @@ def _delete_cascade(self):
             try:
                 delete_count += self.delete_quick(get_count=True)
             except IntegrityError as error:
-                match = foregn_key_error_regexp.match(error.args[0])
-                assert match is not None, "foreign key parsing error"
+                match = (foreign_key_full_error_regexp.match(error.args[0]) or
```

Comment:
you can use a single regexp to match both types of error messages by making some tokens optional.

Reply:
Ah of course! An easy fix to ensure that it is only parsed once. Just updated this to address this.